### DISCUSSION TITLE
Openmediavault?

### DISCUSSION THREAD

**Discussion by mike4president** [tteck/Proxmox#418] [Ideas]
Hi, I just forum your scripts. Great work man, thanks a lot. Would it be possible for you to make a script for openmediavault as well?

**Comment by brianread108** [comment]
+1 for that

**Comment by nacho-domatix** [comment]
+1

**Reply by tteck** [reply]
VM or LXC?

**Reply by nacho-domatix** [reply]
LXC

**Reply by tteck** [reply]
Done!
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/omv-v4.sh)"
```
admin
openmediavault

**Reply by Koliera** [reply]
Hah, I'd love this, though sadly I don't think there's a way to passthrough the current hdd I use on my OMV vm to the LXC.

**Comment by nacho-domatix** [comment]
Try this:

```
nano /etc/pve/lxc/[id number].conf
mp0: [path to mount],mp=[path to mount in CT]
```
[Source](https://virtualizeeverything.com/2022/05/18/passing-usb-storage-drive-to-proxmox-lxc/)

**Reply by ghost** [reply]
Hello, this is my configuration on the container, but omv is not able to see any disks.
rch: amd64
cores: 2
features: nesting=1
hostname: omv
memory: 1024
mp0: nvme1TB:vm-106-disk-0,mp=/data,size=100G
net0: name=eth0,bridge=vmbr0,hwaddr=76:2E:46:43:26:5D,ip=dhcp,type=veth
onboot: 0
ostype: debian
rootfs: local-lvm:vm-106-disk-0,size=4G
swap: 512
lxc.cgroup2.devices.allow: a
lxc.cap.drop:
lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file

**Reply by tteck** [reply]
That's got your USB serial devices.
Look through this post for ideas to mount a drive https://github.com/tteck/Proxmox/issues/542#issuecomment-1256572254

**Reply by nacho-domatix** [reply]
Hi from Valencia!

Did you follow the instructions of the link I posted before? 

You have to add the disk to Proxmox. Let's say it's mounted in /mnt/pve/fivetb/, you can add any folder of the disk (watch permissions) to any container.

```
root@proxmox:~# cat /etc/pve/lxc/1000.conf 
## NextCloudPi LXC
#### https%3A//tteck.github.io/Proxmox/
#<a href='https%3A//ko-fi.com/D1D7EP4GF'><img src='https%3A//img.shields.io/badge/%E2%98%95-Buy me a coffee-red' /></a>
arch: amd64
cores: 2
features: nesting=1
hostname: nextcloud
memory: 2048
mp0: /mnt/pve/fivetb/ncdata,mp=/var/www/nextcloud/data
net0: name=eth0,bridge=vmbr0,hwaddr=C6:69:EE:68:73:89,ip=dhcp,type=veth
onboot: 1
ostype: debian
rootfs: local-lvm:vm-1000-disk-0,size=16G
swap: 512

```
In the UI should look like that.

![imagen](https://user-images.githubusercontent.com/28144115/192863600-406d9f45-7c86-4858-82dc-491ce2ba2706.png)

Hope this can help.

**Reply by ahmaddxb** [reply]
mp0: /mnt/pve/OMV,mp=/mnt/OMV

I tried this, yes the mount point is there in the root file system of OMV but you can not share it with OMV. It is looking for a hard disk with a file system on it to create shares.

Has anyone actually managed to create share from OMV lxc sucessfully?

**Reply by ahmaddxb** [reply]
Even the rootfs disk doesnt show up in OMV.

![image](https://user-images.githubusercontent.com/43728586/194029450-38f49164-6be7-4915-9f08-25edd10e0969.png)

**Reply by guerocp** [reply]
> Even the rootfs disk doesnt show up in OMV.
> 
> ![image](https://user-images.githubusercontent.com/43728586/194029450-38f49164-6be7-4915-9f08-25edd10e0969.png)

I have the same problem as you, did you figure it out a solution?

**Reply by vincentleclerc** [reply]
Same issue for me.. nothing is showing up after a fresh install, very confusing.

**Reply by LoaneShark** [reply]
> Same issue for me.. nothing is showing up after a fresh install, very confusing.

Try following these steps: https://forum.proxmox.com/threads/openmediavault-installation-in-lxc-with-attached-hw-block-device.29918/page-3#post-210603/

**Reply by vincentleclerc** [reply]
> > Same issue for me.. nothing is showing up after a fresh install, very confusing.
> 
> Try following these steps: https://forum.proxmox.com/threads/openmediavault-installation-in-lxc-with-attached-hw-block-device.29918/page-3#post-210603/

Hey Loane, at what step should we be starting if we ran the script from tteck? You mentioned the install script gets you "part way there". Anything we should skip? Or is the tteck script just useless in its current form since we have to do most of the work manually?

**Comment by LoaneShark** [comment]
Writing here to let people know that I have managed to get this working, the install script gets you part of the way there but it doesn't work in an LXC environment out of the box. In order to actually view and manage disks, you need to follow the steps laid out here: [OpenMediaVault installation in LXC with attached HW block device](https://forum.proxmox.com/threads/openmediavault-installation-in-lxc-with-attached-hw-block-device.29918/page-3#post-210603/). You also have to [create a "stupid" storage mount service](https://forum.proxmox.com/threads/openmediavault-installation-in-lxc-with-attached-hw-block-device.29918/post-164345) in order to manually call /etc/fstab on startup, or your filesystems will be unmounted when you reboot the container. With these changes I managed to get OMV seemingly fully-functional in an LXC.

Note that you do have to manually add the block device addresses to the mount-hook.sh script for each disk you want to expose to the container, or it won't be mounted as /dev/sd*, which OMV is expecting. I'm not sure how much of this if any could be automated with a script, since it may vary depending on your hardware setup.

**Reply by patriot1889** [reply]
Ok, so this worked a treat... until I added more drives and my block devices were given different names. sdb became sdc, etc.

Now, I am wondering if it is possible to do this using something like /disk/by-id & /disk/by-uuid etc. Any ideas on this front?

I also tried creating a symlink using udev for my /dev/sdc to a more predictable name but OMV doesn't want to see that.

**Comment by WildeRNS** [comment]
can someone write step-by-step manual, how to add zfs-storage to LXC environment OMV? thanks!

**Reply by mzurhorst** [reply]
I would appreciate this as well.  I was following those links above in the past days and cannot get this working.

**Comment by laviddichterman** [comment]
The folks at OMV seem nonplussed at this being a LXC. The container the script produces is not very useful. See https://forum.openmediavault.org/index.php?thread/51217-trying-to-access-additional-drives-via-proxmox-with-lxc/  
Something along the lines of a warning on the current script would probably prevent folks (myself included) from going down a rabbit hole.
All that said, these scripts as a collection are an absolutely wonderful resource for folks getting started with proxmox!
